packages/block-editor/src/hooks/duotone.js

```diff
@@ -202,12 +202,18 @@ const withDuotoneControls = createHigherOrderComponent(
 			[ props.clientId ]
 		);
 
+		// CAUTION: code added before the early return will be executed
+		// for all blocks, not just those that support duotone. Code added
+		// above this line should be carefully evaluated for its impact on
+		// performance.
+		if ( ! hasDuotoneSupport || isContentLocked ) {
```

Comment:
I don't think the change in this particular component is needed. Actually this change introduces another bug. 

Basically, if a block that supports duotone changes from locked to unlocked, it will be "remounted" entirely, because the react trees don't correspond in both paths of the condition.

Reply:
Thank you. Let me look at that.

Reply:
✅ Fixed.